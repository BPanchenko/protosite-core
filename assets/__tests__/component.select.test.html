<!doctype html>
<html lang="ru">
	<head>
		<title>[TEST FORM OBJECT] Select Field Form ⁕ Protosite Core</title>

		<link
			rel="preload"
			href="http://assets.protosite.rocks/core/select-field.css"
			as="style"
		>
		<link
			rel="preload"
			href="http://assets.protosite.rocks/fonts/iconic/iconic-sm.ttf"
			as="font"
			type="font/ttf"
			crossorigin
		>

		<link
			href="http://assets.protosite.rocks/fonts/iconic/stylesheet.css"
			rel="stylesheet"
			type="text/css"
		>
		<link
			href="http://assets.protosite.rocks/uikit/document.css"
			rel="stylesheet"
			type="text/css"
		>
		<link
			href="http://assets.protosite.rocks/uikit/utilities.css"
			rel="stylesheet"
			type="text/css"
		>
		<link
			href="http://assets.protosite.rocks/uikit/component/panel.css"
			rel="stylesheet"
			type="text/css"
		>
		
		<script src="../component.select-field.mjs" type="module"></script>
		<style>
			.u-grid {
				display: grid;
				grid-template-columns: auto auto;
				gap: 8vmin;
				justify-content: space-around;
				align-items: center;
			}

			.c-panel:first-of-type .u-grid {
				block-size: 22dvh;
				margin-block-end: 8dvh;
			}
		</style>
	</head>
	<body class="t-document">
		<main class="u-container" data-testid="container">
			<form enctype="text/plain" data-testid="main-form">
				<section class="c-panel s-clean">
					<div class="u-grid">
						<c-select-field
							aria-label="Выбран:"
							aria-placeholder="Плод на выбор"
							autofocus
							data-name="custom-field-fruit"
							data-testid="target-component"
						>
							<div role="option" value="kumquat">Кумкват</div>
							<div role="option" value="pseudocydonia"><div>Псевдоайва</div></div>
							<div role="option" value="salacca-wallichiana">Ракум-салакка</div>
							<div role="option" value="mammea-americana">Антильский абрикос</div>
							<div role="option" value="casimiroa-edulis">Белая сапота</div>
						</c-select-field>
					</div>
				</section>
				<section class="c-panel">
					<output data-testid="output"></output>
				</section>
				<section class="c-panel s-clean u-text-center">
					<button data-testid="submitter" type="submit">Сохранить</button>
					<button type="reset">Очистить</button>
				</section>
			</form>
			<script>
				const $component = document.querySelector('[data-testid=target-component]')
				const $form = document.querySelector('[data-testid=main-form]')
				const $output = document.querySelector('[data-testid=output]')
				const $submitter = $form.querySelector('[data-testid=submitter]')

				$form.addEventListener('submit', event => {
					event.preventDefault()
					const $data = document.createElement('pre')
					const formData = new FormData($form, $submitter)

					console.log($form.elements)
					console.log(Array.from(formData.keys()), Array.from(formData.values()))

					for (const [key, value] of formData) {
						$data.textContent += `${key}: ${value}\n`;
					}
					$output.prepend($data)
				})
			</script>
		</main>
	</body>
</html>
