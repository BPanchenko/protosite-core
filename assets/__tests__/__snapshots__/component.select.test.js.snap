// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField Page Markup 1`] = `
"<!DOCTYPE html><html lang="ru"><head>
		<title>[TEST FORM OBJECT] Select Field Form ⁕ Protosite Core</title>

		<link rel="preload" href="http://assets.protosite.rocks/core/component.select.css" as="style">
		<link rel="preload" href="http://assets.protosite.rocks/fonts/iconic/iconic-sm.ttf" as="font" type="font/ttf" crossorigin="">

		<link href="http://assets.protosite.rocks/fonts/iconic/stylesheet.css" rel="stylesheet" type="text/css">
		<link href="http://assets.protosite.rocks/uikit/elements.css" rel="stylesheet" type="text/css">
		<link href="http://assets.protosite.rocks/uikit/utilities.css" rel="stylesheet" type="text/css">
		<link href="http://assets.protosite.rocks/uikit/component/panel.css" rel="stylesheet" type="text/css">
		
		<script src="../component.select.mjs" type="module"></script>
		<style>
			.u-container {
				padding-block-start: 25dvh;
			}

			.u-grid {
				display: grid;
				grid-template-columns: 1fr 1fr;
				gap: 8vmin;
				justify-content: space-around;
				align-items: center;

				.c-panel {
					text-align: center;
				}

				.c-panel:not([aria-hidden=true]) + .c-panel {
					margin-top: unset;
				}
			}


			hr:first-of-type {
				margin-block-start: 8dvh;
			}
		</style>
	</head>
	<body class="t-document">
		<main class="u-container" data-testid="container">
			<form enctype="text/plain" data-testid="main-form">
				<div class="u-grid">
					<section class="c-panel s-clean">
						<c-select aria-label="Выбран:" aria-placeholder="Фрукт на выбор..." name="fruit-name" aria-atomic="true" aria-expanded="false" exportparts="selectedcontent listbox" id="" role="combobox" tabindex="0" aria-owns="option-ota95c option-14t21m option-k5178x option-bp75xb option-85aa4k option-p8ypey option-h526ab option-xch9hy option-amomaa option-t5b3x4 option-x2b2mh option-1ammce option-7o6e5t option-b1y53k option-eoaax5 option-opk5hc option-tay812 option-bopab2 option-bxppt4 option-eokc8o option-t37pk7 option-863527 option-xt5768 option-k16bh4 option-tb4584 option-mat9k9 option-m766m6 option-o96kt8 option-3cpemt option-24axak option-243ob4 option-52x8ma option-7k2ke6 option-9972ck option-bkchb7">
							<div role="option" value="Durian" aria-selected="false" id="option-ota95c">Дуриан</div>
							<div role="option" value="Mangosteen" aria-selected="false" id="option-14t21m">Мангостин</div>
							<div role="option" value="Lychee" aria-selected="false" id="option-k5178x">Личи</div>
							<div role="option" value="Jackfruit" aria-selected="false" id="option-bp75xb">Джекфрут</div>
							<div role="option" value="Papaya" aria-selected="false" id="option-85aa4k">Папайя</div>
							<div role="option" value="Mango" aria-selected="false" id="option-p8ypey">Манго</div>
							<div role="option" value="Rambutan" aria-selected="false" id="option-h526ab">Рамбутан</div>
							<div role="option" value="Longan" aria-selected="false" id="option-xch9hy">Лонган</div>
							<div role="option" value="Carambola" aria-selected="false" id="option-amomaa">Карамбола</div>
							<div role="option" value="Pitaya" aria-selected="false" id="option-t5b3x4">Питайя</div>
							<div role="option" value="Ackee" aria-selected="false" id="option-x2b2mh">Аки</div>
							<div role="option" value="Chompoo" aria-selected="false" id="option-1ammce">Чомпу</div>
							<div role="option" value="Sugar Apple" aria-selected="false" id="option-7o6e5t">Сахарное яблоко</div>
							<div role="option" value="Sapodilla" aria-selected="false" id="option-b1y53k">Саподилла</div>
							<div role="option" value="Guava" aria-selected="false" id="option-eoaax5">Гуава</div>
							<div role="option" value="Kiwano" aria-selected="false" id="option-opk5hc">Кивано</div>
							<div role="option" value="Bael" aria-selected="false" id="option-tay812">Баиль</div>
							<div role="option" value="Salak" aria-selected="false" id="option-bopab2">Салак</div>
							<div role="option" value="Bambangan" aria-selected="false" id="option-bxppt4">Бам-балан</div>
							<div role="option" value="Ambarella" aria-selected="false" id="option-eokc8o">Амбарелла</div>
							<div role="option" value="Citron" aria-selected="false" id="option-t37pk7">Цитрон</div>
							<div role="option" value="Kumquat" aria-selected="false" id="option-863527">Кумкват</div>
							<div role="option" value="Marula" aria-selected="false" id="option-xt5768">Марула</div>
							<div role="option" value="Tamarind" aria-selected="false" id="option-k16bh4">Тамаринд</div>
							<div role="option" value="Miracle Fruit" aria-selected="false" id="option-tb4584">Магический фрукт</div>
							<div role="option" value="Noni" aria-selected="false" id="option-mat9k9">Нони</div>
							<div role="option" value="Soursop" aria-selected="false" id="option-m766m6">Сметанное яблоко</div>
							<div role="option" value="Naranjilla" aria-selected="false" id="option-o96kt8">Наранхилла</div>
							<div role="option" value="Mamey" aria-selected="false" id="option-3cpemt">Мамея</div>
							<div role="option" value="Pepino Dulce" aria-selected="false" id="option-24axak">Пепино</div>
							<div role="option" value="Milky Fruit" aria-selected="false" id="option-243ob4">Милкифрут</div>
							<div role="option" value="Marang" aria-selected="false" id="option-52x8ma">Маранг</div>
							<div role="option" value="Cupuasu" aria-selected="false" id="option-7k2ke6">Купуасу</div>
							<div role="option" value="Curuba" aria-selected="false" id="option-9972ck">Куруба</div>
							<div role="option" value="Jabuticaba" aria-selected="false" id="option-bkchb7">Жаботикаба</div>
						</c-select>
					</section>
					<section class="c-panel">
						<c-select aria-label="Выбран:" aria-placeholder="Цветок на выбор..." name="flower-name" aria-atomic="true" aria-expanded="false" exportparts="selectedcontent listbox" id="" role="combobox" tabindex="0" aria-owns="option-372y22 option-oxp9a8 option-xoph34 option-142yko option-a234yc option-22kyoe option-x2khhc option-8to1c2 option-a2b129 option-73pt9x option-hkxo17 option-5t71bc option-e99h85 option-ehphe9 option-xmc233 option-ocy6ty option-844k6p option-oo33b1 option-y5c895 option-m6ocxt option-61by23 option-6o9h8k option-9bkoem option-cthme2 option-113t66 option-yayxt1 option-457h43 option-b3yk27 option-h3e5k7 option-c7eoxa option-ha472k option-k364b6 option-e81394 option-h2h1tt option-43xt87 option-76296y option-mpt8m4 option-2c2eh1 option-ye4ch2 option-ae77a1">
							<div role="option" value="Anemone" aria-selected="false" id="option-372y22">Анемон</div>
							<div role="option" value="Allium" aria-selected="false" id="option-oxp9a8">Аллиум</div>
							<div role="option" value="Amaryllis" aria-selected="false" id="option-xoph34">Амариллис</div>
							<div role="option" value="Pineapple" aria-selected="false" id="option-142yko">Ананас</div>
							<div role="option" value="Aster" aria-selected="false" id="option-a234yc">Астер</div>
							<div role="option" value="Astilba" aria-selected="false" id="option-22kyoe">Астильба</div>
							<div role="option" value="Brassica" aria-selected="false" id="option-x2khhc">Брассика</div>
							<div role="option" value="Bouvardia" aria-selected="false" id="option-8to1c2">Бувардия</div>
							<div role="option" value="Wanda" aria-selected="false" id="option-a2b129">Ванда</div>
							<div role="option" value="Veronika" aria-selected="false" id="option-73pt9x">Вероника</div>
							<div role="option" value="Viburnum" aria-selected="false" id="option-hkxo17">Вибурнум</div>
							<div role="option" value="Genista" aria-selected="false" id="option-5t71bc">Гениста</div>
							<div role="option" value="Hydrangea" aria-selected="false" id="option-e99h85">Гортензия</div>
							<div role="option" value="Hypericum" aria-selected="false" id="option-ehphe9">Гиперикум</div>
							<div role="option" value="Hyacinth" aria-selected="false" id="option-xmc233">Гиацинт</div>
							<div role="option" value="Delphinium" aria-selected="false" id="option-ocy6ty">Дельфиниум</div>
							<div role="option" value="Calla Lilies" aria-selected="false" id="option-844k6p">Каллы</div>
							<div role="option" value="Kohia" aria-selected="false" id="option-oo33b1">Кохия</div>
							<div role="option" value="Clematis" aria-selected="false" id="option-y5c895">Клематис</div>
							<div role="option" value="Leukadendron" aria-selected="false" id="option-m6ocxt">Леукадендрон</div>
							<div role="option" value="Leukospermum" aria-selected="false" id="option-61by23">Леукоспермум</div>
							<div role="option" value="Limonium" aria-selected="false" id="option-6o9h8k">Лимониум</div>
							<div role="option" value="Mattiola" aria-selected="false" id="option-9bkoem">Маттиола</div>
							<div role="option" value="Mimosa" aria-selected="false" id="option-cthme2">Мимоза</div>
							<div role="option" value="Ozothamnus" aria-selected="false" id="option-113t66">Озотамнус</div>
							<div role="option" value="Ornithogalum" aria-selected="false" id="option-yayxt1">Орнитогалум</div>
							<div role="option" value="Skimiya" aria-selected="false" id="option-457h43">Скимия</div>
							<div role="option" value="Solidago" aria-selected="false" id="option-b3yk27">Солидаго</div>
							<div role="option" value="Strelitzia" aria-selected="false" id="option-h3e5k7">Стрелиция</div>
							<div role="option" value="Sunflower" aria-selected="false" id="option-c7eoxa">Подсолнух</div>
							<div role="option" value="Protea" aria-selected="false" id="option-ha472k">Протея</div>
							<div role="option" value="Wheat" aria-selected="false" id="option-k364b6">Пшеница</div>
							<div role="option" value="Ranunculus" aria-selected="false" id="option-e81394">Ранункулюс</div>
							<div role="option" value="Thanacetum" aria-selected="false" id="option-h2h1tt">Танацетум</div>
							<div role="option" value="Tweedy" aria-selected="false" id="option-43xt87">Твидия</div>
							<div role="option" value="Trachelium" aria-selected="false" id="option-76296y">Трахелиум</div>
							<div role="option" value="Celosia" aria-selected="false" id="option-mpt8m4">Целозия</div>
							<div role="option" value="Freesia" aria-selected="false" id="option-2c2eh1">Фрезия</div>
							<div role="option" value="Heleborus" aria-selected="false" id="option-ye4ch2">Хелеборус</div>
							<div role="option" value="Eringium" aria-selected="false" id="option-ae77a1">Эрингиум</div>
						</c-select>
					</section>
				</div>
				<hr>
				<output data-testid="output"></output>
				<hr>
				<section class="c-panel s-clean u-text-center">
					<button data-testid="submitter" type="submit">Вывести данные формы</button>
					<button type="reset">Сбросить форму</button>
				</section>
			</form>
			<script>
				const $component = document.querySelector('[data-testid=target-component]')
				const $form = document.querySelector('[data-testid=main-form]')
				const $output = document.querySelector('[data-testid=output]')
				const $submitter = $form.querySelector('[data-testid=submitter]')

				$form.addEventListener('submit', event => {
					event.preventDefault()
					const $data = document.createElement('pre')
					const formData = new FormData($form, $submitter)

					console.log($form.elements)
					console.log(Array.from(formData.keys()), Array.from(formData.values()))

					for (const [key, value] of formData) {
						$data.textContent += \`\${key}: \${value}\\n\`;
					}
					$output.prepend($data)
				})
			</script>
		</main>
	

</body></html>"
`;
